### ISSUE TITLE
Serilog fails to write JSON message correctly if it contains the "}" character inside a value

### ISSUE THREAD

**Issue by IustinMicu** [serilog/serilog#1631]
**Description**
When a JSON log contains the **}** character inside a field value, the message is not written correctly to the sink.
To be more specific, the last parenthesis (this one: **}**) of the json log is not printed.

I discovered this issue when Filebeat failed to parse these log messages.
The issue is reproducible with Console and File sinks so I suppose it's not a matter on the sinks but on the core logger itself.

**Reproduction**
- Json log message:
```text
"{\"eventData\":{\"message\":\"Log message plus some details serialized as json: { details_serialized_as_json }\"}}"
```
- Serilog configuration:
```c#
 _logger = new LoggerConfiguration()
                .WriteTo.File(
                    @"C:\Logs\test.log",
                    outputTemplate: "{Message}{NewLine}",
                    rollOnFileSizeLimit: true,
                    formatProvider:null,
                    fileSizeLimitBytes: 10000000,
                    retainedFileCountLimit: 31,
                    shared: true)
                .MinimumLevel.Verbose()
                .WriteTo.Console()
                .CreateLogger();
```

- Usage:
```C#
var message = "{\"eventData\":{\"message\":\"Log message plus some details serialized as json: { details_serialized_as_json }\"}}";
_logger.Debug(message);
```

**Expected behavior**
In the console and in the `C:\Logs\test.log` file I should see this log:
```json
{"eventData":{"message":"Log message plus some details serialized as json: { details_serialized_as_json }"}}
```
**Actual behavior**
```json
{"eventData":{"message":"Log message plus some details serialized as json: { details_serialized_as_json }"}
```
_The last parenthesis is missing._

**Relevant package, tooling and runtime versions**
- Platform: windows x64, 
- .NET: .NET Framework 4.6.1
- Serilog: 2.10.0
- Serilog.Sinks.Console: 4.0.0
- Serilog.Sinks.File: 5.0.0

**Additional context**
Using 
```c#
var message = "{\"eventData\":{\"message\":\"Log message plus some details serialized as json: { details_serialized_as_json }\"}}";
System.IO.File.WriteAllText(@"C:\Logs\test.log", message);
``` 
prints the message correctly.

**Comment by IustinMicu** [comment]
In the meantime I found this similar issue: https://github.com/serilog/serilog-sinks-file/issues/102, but I wouldn't consider this one as duplicate because I am not using string interpolation.
My log message is a valid Json string.

First of all, the fix proposed in that other issue worked for me.
On the other hand I think that Serilog should not treat a log message as a template all the time it sees brackets inside the message, especially if the message string is the only argument provided to the logging method (thus indicating that the user didn't want to treat the message as a template).

Having a Json message that contains another serialized Json string inside of it, results in a valid Json message, so it's a valid scenario and should be supported by Serilog.

**Comment by bartelink** [comment]
>My log message is a valid Json string.

that's not relevant. Is it a valid `String.Format` format string?

TL;DR if you want to log a message with embedded braces that do not represent holes in a message template, you need to do as you've done:

    _logger.Debug("{Message}", message);

In other words, IMO unless I'm truly missing something, this is working as designed and should hence be closed

**Comment by IustinMicu** [comment]
I don't want to log a "format string".
I don't want my log message to be modified at all by Serilog.
I want the log message that I send to Serilog to be written to the sink exactly as I sent it. Serilog should not care what's inside that string (json, plain text, xml, etc.) until I tell Serilog that I want it to do that.
I formatted myself the log message beforehand, so I don't need Serilog for that purpose. It's very bad if Serilog forces me to always send a template string.
Serilog's main job is to write logs to a sink, right? Extra features like templating should be something that I can opt out from.

**Comment by bartelink** [comment]
If you truly need low level stuff (I once did, but I think it's overkill for what you are talking about), see https://stackoverflow.com/a/49639001/11635

I am not trying to fight you on anything, but the bottom line is that the first argument to Serliog's Log methods _is a message template_
This explains stuff 'going wrong'

Regardless of your opinion or mine, the ship has sailed, and there's really not much to be won by debating here

**Comment by IustinMicu** [comment]
My bad if I sounded a bit too harsh. I didn't mean to argue!
Apparently I had wrong expectations..

If you say it's not possible to change the way it works then we can close this issue.
At least I understand how it is intended to be used.
Thank you very much for clarifying it :)

Best regard!

**Comment by bartelink** [comment]
Yes, I'm pretty confident it's working as designed, even though I personally am not an expert in this area of Serilog's design or functionality.

My main intent here is not to provide authoritative answers, but to assist in helping with resolving low-hanging issues so maintainer's time is maximally spent on issues that they are uniquely placed to address

(IIRC I commented on a similar SO question in recent times; can't locate it now)

**Comment by ezyuzin** [comment]
> My bad if I sounded a bit too harsh. I didn't mean to argue! Apparently I misunderstood the Serilog usage..
> 
> If you say it's not possible to change the way it works then we can close this issue. Maybe you can consider in the future to extend the current API with some _Raw_ methods (DebugRaw, InformationRaw, etc.) that print the message as it is.
> 
> Anyways, at least for now I understand how it is intended to be used. Thank you very much for clarifying it :)
> 
> Best regard!

simple extension for log raw message:

    public static void Info(this ILogger log, string message)
    {
      if (log.IsEnabled(LogLevel.Information)) {
        // костыли для форматтера Serilog чтобы печатал сообщения как есть в лог, без подстановки
        log.LogInformation("{0}", message);
      }
    }

**Comment by nblumhardt** [comment]
Thanks @ezyuzin 👍 

Another similar option:

```csharp
public static void Literal(this ILogger logger, LogEventLevel level, Exception? exception, string message)
{
    if (logger.IsEnabled(level))
    {
        logger.Write(new LogEvent(
            DateTimeOffset.Now,
            level,
            exception,
            new MessageTemplate([new TextToken(message)]),
            [],
            Activity.Current?.TraceId ?? default,
            Activity.Current?.SpanId ?? default));
    }
}
```

This should have a similar effect, but won't attach an additional "0" property to the event (so will save some allocations and formatting time).

**Comment by ezyuzin** [comment]
> This should have a similar effect, but won't attach an additional "0" property to the event (so will save some allocations and formatting time).

In my case it is not accepted solution/
I'am use Microsoft.Extensions.Logging abstractions from Microsoft DI for logging and it not contains such signature, need add  reference directly to Serilog Logger or use Serilog abstractions

:(
